packages/common-server/src/parser.ts

```diff
@@ -129,6 +145,31 @@ export class SchemaParserV2 extends ParserBaseV2 {
     return collector;
   }
 
+  static createSchema(
+    opts: (SchemaOpts | SchemaRaw) & { vault: DVault }
+  ): SchemaProps {
+    this.validateSchemaObjOptsOnlyUseAllowedKeys(opts);
+
+    return SchemaUtils.create(opts);
+  }
+
+  private static validateSchemaObjOptsOnlyUseAllowedKeys(
```

Comment:
we currently use [ajv](https://ajv.js.org/guide/typescript.html) library for pods to validate and error on schema. you might want to use this instead of creating our own (note: you'll need to add `ajv` to `common-server` as a dependency since its currently only used in `pods-core`)

Reply:
see how we're using it [here](https://github.com/dendronhq/dendron-site/blob/dev/vault/pkg.pods-core.ref.validation.md#L17:L17)

Reply:
Added AJV for the bulk the bulk of the schema validation.

NOTE: Added a brand new dependency ajv-errors (https://www.npmjs.com/package/ajv-errors) as part of this. It has MIT license and 10Milllion weekly downloads. This is used to format AJV errors in a user friendly way. 

Still kept some of the manual validation like validating that only the top schemas to require ids, since it appears overkill to create another separate AJV schema just for validation of id existence (if there are other cases specific to top level we can refactor into AJV validation).